src/parameters.rs

```diff
@@ -0,0 +1,334 @@
+    US,
+    #[value(name = "EU")]
+    EU,
+    #[value(name = "Staging")]
+    Staging,
+    #[value(name = "Custom")]
+    Custom,
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct ExternalEndpoints {
+    /// Custom endpoint for token renewal. Required with '--environment custom'.
+    #[arg(long)]
+    token_renewal_endpoint: Option<String>,
+
+    /// Custom endpoint for system identity creation. Required with '--environment custom'.
+    #[arg(long, short)]
+    system_identity_creation_endpoint: Option<String>,
+}
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum IdentityType {
+    /// Creates an identity whose type is 'secret'
+    /// This type o identity expires
+    ///
+    /// EXAMPLE:
+    /// SystemIdentity {
+    /// id: "2e483fe9",
+    /// name: Some("test1"),
+    /// client_id: "8dbf3d32",
+    /// organization_id: "b961cf81",
+    /// identity_type: L1 {
+    /// client_secret: "AfYFAUjf9",
+    /// credential_expiration: "2025-06-04T19:25:00Z"
+    /// }
+    /// }
+    Secret(SecretArgs),
+    /// Creates an identity whose type is 'private key' or known as a parent identity.
+    /// This type of identity does not expire.
+    ///
+    /// EXAMPLE:
+    /// SystemIdentity(
+    /// id: e5af42f2,
+    /// name: test,
+    /// client_id: 8150a0ee,
+    /// organization_id: b961cf81,
+    /// identity_type: L2(pub_key: LS0tLS1)
+    /// )
+    Key(KeyArgs),
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct KeyArgs {
+    /// Basic information need for auth name, client_id, etc.
+    #[command(flatten)]
+    basic_auth_args: BasicAuthArgs,
+
+    /// Add the access token for identity creation, only bearer token type is accepted
+    #[arg(long)]
+    bearer_access_token: String,
+
+    /// Options for configuring the output destination (required for Key identity)
+    #[command(flatten)]
+    output_options: OutputDestinationArgs,
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct SecretArgs {
+    /// Basic information need for auth name, client_id, etc.
+    #[command(flatten)]
+    basic_auth_args: BasicAuthArgs,
+
+    /// Add the access token for identity creation, only bearer token type is accepted
+    #[arg(long)]
+    bearer_access_token: String,
+}
+
+#[derive(ValueEnum, Clone, Debug, PartialEq)]
+pub enum OutputPlatformChoice {
+    #[value(name = "local-file")]
+    LocalFile,
+}
+
+#[derive(Args, Debug, Clone)]
+pub struct OutputDestinationArgs {
+    /// Platform for the output of the generated key or resource.
+    #[arg(long, short, value_enum)]
+    output_platform: OutputPlatformChoice,
+
+    /// File path where the private key output will be saved (required if --output-platform=local-file).
+    #[arg(long, short)]
+    output_local_path: Option<PathBuf>,
+}
+
+pub fn create_metadata_for_token_retrieve(
+    client_id: String,
+    environment: Environments,
+    auth_args: &AuthInputArgs,
+    endpoints: ExternalEndpoints,
+) -> Result<SystemTokenCreationMetadata, Box<dyn std::error::Error>> {
```

Comment:
I'd add more context to the error. If `select_environment` failed (with an invalid url for `Custom`) we would end-up with the raw error with no context.

We could either type the error (as we do in the rest of the code) or, at lest, include more details with `map_error`.